### DISCUSSION TITLE
How can I detect sensitive information and blur/unblur it?

### DISCUSSION THREAD

**Discussion by unKnownNG** [mindee/doctr#1652] [Q&A]
Hello Developers,

I am currently working in a project for a contest from a company. The contest is to create a secure file transfer system for sharing confidential documents and other stuffs.  

My task is to create a program that can recognize the sensitive information(Name, ID, Address, etc) from a png or jpg document and blur the sensitive information. Also after the file transfer, when the document is successfully transferred to the host computer, I have to clear up the blurred sensitive content. 

I can use docTR to identify the content in the document, after identifying what technology should I proceed with and how can I blur and clear up the blurred content?

Should I use machine learning algorithm to detect the sensitive information? If yes what algorithm should I use?
Considering I am good at Python programming language and have a strong understating in difficulty topics like OOPs and various python libraries.

I am just a student aspiring to become a dev like you, I need your guidance to compete in this competition .

Thanking you.

**Comment by felixdittrich92** [comment]
Hi @unKnownNG :wave:,

For a really quick prototype you could try the following:

- OCR with docTR (`.export()` to get all necessary information / `.render()` with `resolve_blocks=False` to get the full text line based)
- NER to get the entities you want to blur later on : https://github.com/urchade/GLiNER
- map the results to the docTR results to get the coordinates for each entity
- blur the boxes / polygons in the original image

Another option would be to train a custom KIE model but this requires an larger annotated dataset so i suggest to try the faster approach first :)

Best regards,
Felix
